python_modules/dagster/dagster/_core/blueprints/blueprint_assets_definition.py

```diff
@@ -0,0 +1,51 @@
+from abc import abstractmethod
+from typing import AbstractSet, Any, Mapping, Optional, Sequence
+
+from dagster._core.blueprints.blueprint import Blueprint, BlueprintDefinitions
+from dagster._core.definitions.asset_spec import AssetSpec
+from dagster._core.definitions.assets import unique_id_from_asset_and_check_keys
+from dagster._core.definitions.decorators.asset_decorator import multi_asset
+from dagster._core.execution.context.compute import AssetExecutionContext
+from dagster._model import DagsterModel
+
+
+class AssetSpecModel(DagsterModel):
+    key: str
+    deps: Sequence[str] = []
+    description: Optional[str] = None
+    metadata: Mapping[str, Any] = {}
+    group_name: Optional[str] = None
+    skippable: bool = False
+    code_version: Optional[str] = None
+    owners: Sequence[str] = []
+    tags: Mapping[str, str] = {}
+
+    def to_asset_spec(self) -> AssetSpec:
+        return AssetSpec(**self.__dict__)
+
+
+class BlueprintAssetsDefinition(Blueprint):
+    """A blueprint that produces an AssetsDefinition."""
+
+    assets: Sequence[AssetSpecModel]
```

Comment:
In general I think we should be having folks exposing very bespoke yaml schemas and creating asset specs programmatically in python, but we can iterate on that as we go.

Reply:
Yes, definitely